### PR TITLE
Overhaul XML handler docs

### PR DESCRIPTION
**Pull request by Girgias**
This also removes the last remaining usage of the ``&note.func-callback;`` language snippet.

### CODE REVIEW

File: language-snippets.ent [hunk 1]
```diff
@@ -4258,6 +4258,32 @@ xmlns="http://docbook.org/ns/docbook"><simpara>This function has been
...
+  </para>
+ </listitem>
+</varlistentry>'>
+
+<!ENTITY xml.handler.description '<para xmlns="http://docbook.org/ns/docbook">
+ If &null; or an empty string is passed, the handler is reset to its default state.
+</para>
+<para xmlns="http://docbook.org/ns/docbook">
+ If <parameter>handler</parameter> is a <type>string</type>,
+ it can be the name of a method of an object set with
```
Comment (nielsdos) [inline language-snippets.ent:4275]: I think this wording is not optimal.
The note about array(obj, method) was removed, and this `can` is suggestive that this is the usual way to go about it (and no other way is possible).
Not sure how to improve it though.

File: reference/xml/functions/xml-set-element-handler.xml [hunk 2]
```diff
@@ -16,118 +16,89 @@
   </methodsynopsis>
   <para>
    Sets the element handler functions for the XML <parameter>parser</parameter>.
-   <parameter>start_handler</parameter> and
-   <parameter>end_handler</parameter> are strings containing
-   the names of functions that must exist when <function>xml_parse</function>
-   is called for <parameter>parser</parameter>.
+   <parameter>start_handler</parameter> is called when a new XML element is
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-element-handler.xml:19]: Due to the fact that the newline is not rendered in the HTML output, it looks like `parser.start_handler` is called (...), which I find confusing tbh.

File: reference/xml/functions/xml-set-end-namespace-decl-handler.xml [hunk 3]
```diff
@@ -31,44 +31,20 @@
...
-       function that must exist when <function>xml_parse</function> is called
-       for <parameter>parser</parameter>.
-      </para>
-      <para>
-       The function named by <parameter>handler</parameter> must accept
-       two parameters, and should return an integer value. If the
-       value returned from the handler is &false; (which it will be if no
-       value is returned), the XML parser will stop parsing and
-       <function>xml_get_error_code</function> will return
-       <constant>XML_ERROR_EXTERNAL_ENTITY_HANDLING</constant>.
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-end-namespace-decl-handler.xml:56]: I wonder how long this wrong remark has been in the manual

**Review by nielsdos — COMMENTED** [review]
Changes look good.
I just have to check whether false can actually be passed, and if so, in which cases...

File: reference/xml/functions/xml-set-external-entity-ref-handler.xml [hunk 4]
```diff
@@ -23,47 +23,23 @@
...
-        <methodname><replaceable>handler</replaceable></methodname>
+        <type>bool</type><methodname><replaceable>handler</replaceable></methodname>
         <methodparam><type>XMLParser</type><parameter>parser</parameter></methodparam>
-        <methodparam><type>string</type><parameter>open_entity_names</parameter></methodparam>
-        <methodparam><type>string</type><parameter>base</parameter></methodparam>
-        <methodparam><type>string</type><parameter>system_id</parameter></methodparam>
-        <methodparam><type>string</type><parameter>public_id</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>open_entity_names</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>base</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>system_id</parameter></methodparam>
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-external-entity-ref-handler.xml:38]: system_id can never be false

File: reference/xml/functions/xml-set-external-entity-ref-handler.xml [hunk 5]
```diff
@@ -23,47 +23,23 @@
...
+       The signature of the handler must be:
        <methodsynopsis>
-        <methodname><replaceable>handler</replaceable></methodname>
+        <type>bool</type><methodname><replaceable>handler</replaceable></methodname>
         <methodparam><type>XMLParser</type><parameter>parser</parameter></methodparam>
-        <methodparam><type>string</type><parameter>open_entity_names</parameter></methodparam>
-        <methodparam><type>string</type><parameter>base</parameter></methodparam>
-        <methodparam><type>string</type><parameter>system_id</parameter></methodparam>
-        <methodparam><type>string</type><parameter>public_id</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>open_entity_names</parameter></methodparam>
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-external-entity-ref-handler.xml:36]: open_entity_names cannot be false

File: reference/xml/functions/xml-set-character-data-handler.xml [hunk 6]
```diff
@@ -23,46 +23,25 @@
...
-      <para>
-       The function named by <parameter>handler</parameter> must accept
-       two parameters:
+       The signature of the handler must be:
        <methodsynopsis>
-        <methodname><replaceable>handler</replaceable></methodname>
+        <type>void</type><methodname><replaceable>handler</replaceable></methodname>
         <methodparam><type>XMLParser</type><parameter>parser</parameter></methodparam>
-        <methodparam><type>string</type><parameter>data</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>data</parameter></methodparam>
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-character-data-handler.xml:36]: Can't be false

File: reference/xml/functions/xml-set-notation-decl-handler.xml [hunk 7]
```diff
@@ -35,50 +35,28 @@
...
+       The signature of the handler must be:
        <methodsynopsis>
-        <methodname><replaceable>handler</replaceable></methodname>
+        <type>void</type><methodname><replaceable>handler</replaceable></methodname>
         <methodparam><type>XMLParser</type><parameter>parser</parameter></methodparam>
-        <methodparam><type>string</type><parameter>notation_name</parameter></methodparam>
-        <methodparam><type>string</type><parameter>base</parameter></methodparam>
-        <methodparam><type>string</type><parameter>system_id</parameter></methodparam>
-        <methodparam><type>string</type><parameter>public_id</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>notation_name</parameter></methodparam>
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-notation-decl-handler.xml:48]: Can't be false

File: reference/xml/functions/xml-set-notation-decl-handler.xml [hunk 8]
```diff
@@ -35,50 +35,28 @@
...
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>base</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>system_id</parameter></methodparam>
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-notation-decl-handler.xml:50]: system_id can never be false

File: reference/xml/functions/xml-set-processing-instruction-handler.xml [hunk 9]
```diff
@@ -21,83 +21,65 @@
...
-       The function named by <parameter>handler</parameter> must accept
-       three parameters:
+       The signature of the handler must be:
        <methodsynopsis>
-        <methodname><replaceable>handler</replaceable></methodname>
+        <type>void</type><methodname><replaceable>handler</replaceable></methodname>
         <methodparam><type>XMLParser</type><parameter>parser</parameter></methodparam>
-        <methodparam><type>string</type><parameter>target</parameter></methodparam>
-        <methodparam><type>string</type><parameter>data</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>target</parameter></methodparam>
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-processing-instruction-handler.xml:60]: Target and data can never be false

File: reference/xml/functions/xml-set-start-namespace-decl-handler.xml [hunk 10]
```diff
@@ -27,45 +27,21 @@
...
-       <constant>XML_ERROR_EXTERNAL_ENTITY_HANDLING</constant>.
+       The signature of the handler must be:
        <methodsynopsis>
-        <methodname><replaceable>handler</replaceable></methodname>
+        <type>void</type><methodname><replaceable>handler</replaceable></methodname>
         <methodparam><type>XMLParser</type><parameter>parser</parameter></methodparam>
-        <methodparam><type>string</type><parameter>prefix</parameter></methodparam>
-        <methodparam><type>string</type><parameter>uri</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>prefix</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>uri</parameter></methodparam>
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-start-namespace-decl-handler.xml:41]: uri can't be false

File: reference/xml/functions/xml-set-unparsed-entity-decl-handler.xml [hunk 11]
```diff
@@ -41,45 +41,24 @@
...
+        <type>void</type><methodname><replaceable>handler</replaceable></methodname>
         <methodparam><type>XMLParser</type><parameter>parser</parameter></methodparam>
-        <methodparam><type>string</type><parameter>entity_name</parameter></methodparam>
-        <methodparam><type>string</type><parameter>base</parameter></methodparam>
-        <methodparam><type>string</type><parameter>system_id</parameter></methodparam>
-        <methodparam><type>string</type><parameter>public_id</parameter></methodparam>
-        <methodparam><type>string</type><parameter>notation_name</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>entity_name</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>base</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>system_id</parameter></methodparam>
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-unparsed-entity-decl-handler.xml:56]: system_id can never be false

File: reference/xml/functions/xml-set-unparsed-entity-decl-handler.xml [hunk 12]
```diff
@@ -41,45 +41,24 @@
...
        <methodsynopsis>
-        <methodname><replaceable>handler</replaceable></methodname>
+        <type>void</type><methodname><replaceable>handler</replaceable></methodname>
         <methodparam><type>XMLParser</type><parameter>parser</parameter></methodparam>
-        <methodparam><type>string</type><parameter>entity_name</parameter></methodparam>
-        <methodparam><type>string</type><parameter>base</parameter></methodparam>
-        <methodparam><type>string</type><parameter>system_id</parameter></methodparam>
-        <methodparam><type>string</type><parameter>public_id</parameter></methodparam>
-        <methodparam><type>string</type><parameter>notation_name</parameter></methodparam>
+        <methodparam><type class="union"><type>string</type><type>false</type></type><parameter>entity_name</parameter></methodparam>
```
Comment (nielsdos) [inline reference/xml/functions/xml-set-unparsed-entity-decl-handler.xml:54]: entity_name cannot be false

**Review by nielsdos — CHANGES_REQUESTED** [review]
False remarks

File: reference/xml/functions/xml-set-element-handler.xml (hunk 2 above)
Comment (Girgias) [reply reference/xml/functions/xml-set-element-handler.xml:19]: Right, I'll split this into a new para tag

File: language-snippets.ent (hunk 1 above)
Comment (Girgias) [reply language-snippets.ent:4275]: Oh I forgot to describe the param if it is a callable >_>

File: reference/xml/functions/xml-set-element-handler.xml [hunk 13]
```diff
@@ -16,118 +16,89 @@
...
+      &xml.handler.description;
       <para>
-       The function named by <parameter>start_handler</parameter>
-       must accept three parameters:
+       The signature of the handler must be:
        <methodsynopsis>
-        <methodname><replaceable>start_element_handler</replaceable></methodname>
+        <type>void</type><methodname><replaceable>start_element_handler</replaceable></methodname>
         <methodparam><type>XMLParser</type><parameter>parser</parameter></methodparam>
         <methodparam><type>string</type><parameter>name</parameter></methodparam>
```
[the remark this answers is not in the document]
Comment (Girgias) [reply reference/xml/functions/xml-set-element-handler.xml:39]: Yes, because it didn't use ``_xml_xmlchar_zval()`` :D

**Pull request merged**